src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -135,6 +135,14 @@ class ClipBam
       out += rec
     }
 
+    // Set the "All" and "Pair" read types
+    metricsMap.foreach { metrics: Map[ReadType, ClippingMetrics] =>
```

Comment:
I keep getting confused by the fact that `metricsMap` is actually an `Option[Map[A,B]]`.  How would you feel about simplifying things by having the declaration be:

`val metricsMap = if (this.metrics.isEmpty) Map.Empty() else ....`

So that you either have an empty map, or a populated map?  Then where you need to query it you can just call `metricsMap.get()` which will return you an `Option[ClippingMetrics]`?

Reply:
It would also be nice here since you won't have to match it out to `metrics`, which fwiw is shadowing `this.metrics` :/